improver/synthetic_data/generate_metadata.py

```diff
@@ -0,0 +1,239 @@
+        time_period (Optional[float]):
+            The period in hours between the time bounds. This is used to calculate the lower time bound.
+        frt (Optional[datetime.datetime]):
+            Single cube forecast reference time.
+        ensemble_members (Optional[int]):
+            Number of ensemble members. Default 8, unless percentile or probability set to True.
+        leading_dimension (Optional[List[float]]):
+            List of realizations, percentiles or thresholds.
+        percentile (Optional[bool]):
+            Flag to indicate whether the leading dimension is percentile values. If True, a percentile cube is created.
+        probability (Optional[bool]):
+            Flag to indicate whether the leading dimension is threshold values. If True, a probability cube is created.
+        spp__relative_to_threshold (Optional[str]):
+            Value of the attribute "spp__relative_to_threshold" which is required for IMPROVER probability cubes.
+        attributes (Optional[Dict]):
+            Dictionary of additional metadata attributes.
+        resolution (Optional[float]):
+            Resolution of grid (metres or degrees).
+        domain_corner (Optional[Tuple[float, float]]):
+            Bottom left corner of grid domain (y,x) (degrees for latlon or metres for equalarea).
+        npoints (Optional[int]):
+            Number of points along a single axis.
+        height_levels (Optional[List[float]]):
+            List of altitude/pressure levels.
+        pressure (Optional[bool]):
+            Flag to indicate whether the height levels are specified as pressure, in Pa. If False, use height in metres.
+
+    Returns:
+        iris.cube.Cube:
+            Output of set_up_variable_cube(), set_up_percentile_cube() or set_up_probability_cube()
+    """
+    if spatial_grid not in ("latlon", "equalarea"):
+        raise ValueError(
+            "Spatial grid {} not supported. Choose either latlon or equalarea.".format(
+                spatial_grid
+            )
+        )
+
+    if percentile is True and probability is True:
+        raise ValueError(
+            "Only percentile cube or probability cube can be generated, not both."
+        )
+
+    if domain_corner is not None and len(domain_corner) != 2:
+        raise ValueError("Domain corner must be a list or tuple of length 2.")
+
+    if units is None:
+        units = _get_units(name)
+
+    # If time_period specified, create time bounds using time as upper bound
+    if time_period is not None:
+        time_bounds = _create_time_bounds(time, time_period)
+    else:
+        time_bounds = None
+
+    # If resolution not specified, use default for requested spatial grid
+    if resolution is None:
+        resolution = DEFAULT_RESOLUTION[spatial_grid]
+
+    # Create ndimensional array of zeros
+    data = _create_data_array(
+        ensemble_members, leading_dimension, npoints, height_levels
+    )
+
+    # Set up requested cube
+    if percentile:
```

Comment:
Update in commit 915f034491985cbda5cb2a152484abb7444a03ce.

I directly wrote a dictionary instead of using ```{key: value for key, value in locals() if (key in ["data", "units", "time", "time_bounds"])}``` because I felt it made it clearer and the variable names need to be specified anyway. Would there be advantages to creating the dictionary using a for loop?